apps/token-manager/app/src/components/Panels/TokenPanelContent.js

```diff
@@ -17,233 +18,279 @@ import LocalIdentitiesAutoComplete from '../LocalIdentitiesAutoComplete/LocalIde
-    )
+      if (!fieldsData) {
+        return
+      }
 
-    return (
-      <div>
-        <form
+      onUpdateTokens({
+        amount: fieldsData.amount,
+        holder: fieldsData.holder,
+        mode,
+      })
+    },
+    [mode, validateFields, onUpdateTokens]
+  )
+
+  return (
+    <div>
+      <form
+        css={`
+          margin-top: ${3 * GU}px;
+        `}
+        onSubmit={handleSubmit}
+      >
+        <InfoMessage
+          title="Action"
+          text={`This action will ${
+            mode === 'assign'
+              ? 'create tokens and transfer them to the recipient below'
+              : 'remove tokens from the account below'
+          }.`}
+        />
+        <Field
+          label={`${
+            mode === 'assign' ? 'Recipient' : 'Account'
+          } (must be a valid Ethereum address)`}
+          css="height: 62px"
+        >
+          <LocalIdentitiesAutoComplete
+            ref={holderInputRef}
+            value={holderField.value}
+            onChange={updateHolder}
+            wide
+            required
+          />
+        </Field>
+
+        <Field label="Number of tokens">
+          <TextInput.Number
+            value={amountField.value}
+            onChange={handleAmountChange}
+            min={tokenStep}
+            max={amountField.max}
+            step={tokenStep}
+            required
+            wide
+          />
+        </Field>
+        <Button mode="strong" type="submit" disabled={submitDisabled} wide>
+          {mode === 'assign' ? 'Add' : 'Remove'} tokens
+        </Button>
+        <div
           css={`
-            margin-top: ${3 * GU}px;
+            margin-top: ${2 * GU}px;
           `}
-          onSubmit={this.handleSubmit}
         >
-          <InfoMessage
-            title="Action"
-            text={`This action will ${
-              mode === 'assign'
-                ? 'create tokens and transfer them to the recipient below'
-                : 'remove tokens from the account below'
-            }.`}
-          />
-          <Field
-            label={`
-              ${mode === 'assign' ? 'Recipient' : 'Account'}
-              (must be a valid Ethereum address)
-            `}
-            css="height: 62px"
-          >
-            <LocalIdentitiesAutoComplete
-              ref={this._holderInput}
-              value={holderField.value}
-              onChange={this.handleHolderChange}
-              wide
-              required
-            />
-          </Field>
+          {errorMessage && <ErrorMessage message={errorMessage} />}
+          {warningMessage && <WarningMessage message={warningMessage} />}
+        </div>
+      </form>
+    </div>
+  )
+}
 
-          <Field label="Number of tokens">
-            <TextInput.Number
-              value={amountField.value}
-              onChange={this.handleAmountChange}
-              min={minTokenStep}
-              max={amountField.max}
-              disabled={accountMaxed}
```

Comment:
⚠️ We should keep the logic for `accountMaxed`, or find a different mechanism to handle it (e.g. handling it during `updateAmount()`).

Right now, the form doesn't stop you from trying to assigning or removing too many tokens.

Reply:
Done in https://github.com/aragon/aragon-apps/pull/1046/commits/5535c0b17d9e9b4970a08680f60679366ed91383